libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp

```diff
@@ -30,6 +31,74 @@ bool AP_BattMonitor_SMBus::get_cycle_count(uint16_t &cycles) const
     return true;
 }
 
+#if HAL_SMART_BATT_LOG_ENABLED
+// return true if serial number can be provided and fills in serial number argument
+bool AP_BattMonitor_SMBus::get_serial_number(char *serial_number, uint8_t buflen) const
```

Comment:
I don't think a helper that does the snprintf is really worth it. The caller should probably handle this directly as needed, but I guess if you split it into 2 places then keeping this is maybe acceptable?

Reply:
It doesn't help at all. Getting rid of it saved 70ish-bytes